apps/typegpu-docs/src/content/examples/simple/probability/plotter.ts

```diff
@@ -0,0 +1,436 @@
+          requestAnimationFrame(animateTransition);
+        } else {
+          resolve();
+        }
+      };
+      requestAnimationFrame(animateTransition);
+    });
+  }
+
+  #prepareGeometricData(samples: d.v3f[], prng: GeometricPRNG): GeometricData {
+    const data: GeometricData = {
+      count: this.#count,
+      ids: new Uint32Array(this.#count),
+      positionsX: new Float64Array(this.#count),
+      positionsY: new Float64Array(this.#count),
+      positionsZ: new Float64Array(this.#count),
+      sizes: new Float64Array(this.#count),
+      dists: new Float64Array(this.#count),
+    };
+
+    const origin = prng.origin;
+
+    for (let i = 0; i < this.#count; i++) {
+      data.ids[i] = i;
+      data.positionsX[i] = samples[i].x;
+      data.positionsY[i] = samples[i].y;
+      data.positionsZ[i] = samples[i].z;
+      data.sizes[i] = defaultSize;
+      data.dists[i] = Math.sqrt(
+        (data.positionsX[i] - origin.x) ** 2 +
+          (data.positionsY[i] - origin.y) ** 2 +
+          (data.positionsZ[i] - origin.z) ** 2,
+      );
+    }
+
+    return data;
+  }
+
+  #plotGeometric(
+    data: GeometricData,
+    core: MorphCharts.Core,
+    needNewBuffer: boolean,
+  ) {
+    if (this.#transitionBuffer === undefined || needNewBuffer) {
+      this.#transitionBuffer = this.#core.renderer.createTransitionBuffer(
+        data.ids,
+      );
+    }
+
+    this.#core.renderer.transitionBuffers = [this.#transitionBuffer];
+    this.#transitionBuffer.currentBuffer.unitType = MorphCharts.UnitType.sphere;
+    this.#transitionBuffer.currentPalette.colors = this.#palette;
+
+    const scatter = new MorphCharts.Layouts.Scatter(this.#core);
+    scatter.layout(this.#transitionBuffer.currentBuffer, data.ids, {
+      positionsX: data.positionsX,
+      positionsY: data.positionsY,
+      positionsZ: data.positionsZ,
+    });
+    scatter.update(this.#transitionBuffer.currentBuffer, data.ids, {
+      sizes: data.sizes,
+      sizeScaling: 0.1,
+      colors: data.dists,
+    });
+
+    const axes = MorphCharts.Axes.Cartesian3dAxesHelper.create(core, {
+      titleX: 'x',
+      titleY: 'y',
+      titleZ: 'z',
+      labelsX: (value) => {
+        return value.toFixed(1);
+      },
+      labelsY: (value) => {
+        return value.toFixed(1);
+      },
+      labelsZ: (value) => {
+        return value.toFixed(1);
+      },
+    });
+    core.renderer.currentAxes = [
+      core.renderer.createCartesian3dAxesVisual(axes),
+    ];
+  }
+
+  #prepareContinuousData(samples: d.v3f[]): HistogramData {
+    // if we want to animate the transition between the old and new plots,
+    // we need to have the same number of samples,
+    // that's why I clamp them instead of filtering
+    const samplesFiltered = samples.map((sample) => ({
+      ...sample,
+      x: sample.x < defaultMinValue
+        ? defaultMinValue
+        : sample.x > defaultMaxValue
+        ? defaultMaxValue
+        : sample.x,
```

Comment:
```suggestion
      x: clamp(sample.x, defaultMinValue, defaultMaxValue),
```

Also, we usually import `* as std` instead of importing functions directly.